galsim/base.py

```diff
@@ -58,13 +58,21 @@ class GSObject(object):
         >>> conv = galsim.Convolve([gal,psf])
 
     All of these classes are subclasses of GSObject, so you should see those docstrings for
-    more details about how to construct the various profiles.
-
-    Note that most GSObjects have some kind of size specification.  Typically, these would be
-    given in terms of arcsec, with the connection to the pixel size being given in the Pixel
-    class (0.2 arcsec/pixel in the above example).  However, you can have a more complicated
-    relationship between pixel and sky coordinates.  See BaseWCS for more details about
-    how to specify various kinds of world coordinate systems.
+    more details about how to construct the various profiles.  Here we discuss attributes and
+    methods that are common to all GSObjects.
+
+    GSObjects are always defined in sky coordinates.  So all sizes and other linear dimensions
+    should be in terms of some kind of units on the sky, arcsec for instance.  Only later (when
+    they are drawn) is the connection to pixel coordinates established via a pixel scale or WCS.
+    (See the documentation for galsim.BaseWCS for more details about how to specify various kinds
+    of world coordinate systems more complicated than a simple pixel scale.)
+
+    For instance, if you eventually draw onto an image that has a pixel scale of 0.2 arcsec/pixel,
+    then the normal thing to do would be to define your surface brightness profiles in terms of
+    arcsec and then draw with `pixel_scale=0.2`.  However, nowhere in GalSim do we enforce that sky
+    coordinates need to be in arcsec.  If you wanted, you could instead define the sizes of all
```

Comment:
- COSMOSCatalog: Yes, we should at least add some documentation that the objects built by COSMOSCatalog will have their linear dimensions in arcsec.  This seems to be poorly documented at the moment.
- NFW halo: I think the documentation on this is pretty good already that the positions should be given in arcsec.  We have a hook for units in a bunch of places to extend it to allow other units, although we say that only arcsec are implemented so far.  So I think this is ok, but there is room for converting them to use a similar kind of conversion calculation as we do in these classes.
- ChromaticAtmosphere: It would be easy to convert this to take a scale_unit parameter like these classes.  The arcsec is explicitly implemented [here](https://github.com/GalSim-developers/GalSim/blob/master/galsim/chromatic.py#L1113), so it would be very easy to divide by scale_unit instead.  I'll make that change.